### ISSUE TITLE
get method does not work with dns.google(8.8.8.8)

### ISSUE THREAD

**Issue by cattyhouse** [LinkTed/doh-client#39]
build
```sh
cargo b -r --no-default-features
```
command

```sh
./doh-client --get --cache-size 0 --listen-addr 127.0.0.1:9999 --domain dns.google --remote-host 8.8.8.8:443 /etc/ssl/cert.pem

dig @127.0.0.1 -p 9999 ocf.berkeley.edu
```
error message

````
[ERROR doh_client::handler] Could not retrieve DNS response from server: H2 Error: stream error received: not a result of an error
[ERROR doh_client::run] Could not handle request: Could not get response for: Dns { id: 64443, flags: Flags { qr: false, opcode: Query, aa: false, tc: false, rd: true, ra: false, ad: true, cd: false, rcode: NoError }, questions: [Question { domain_name: DomainName([Label("ocf"), Label("berkeley"), Label("edu")]), q_class: IN, q_type: A }], answers: [], authorities: [], additionals: [OPT(OPT { requestor_payload_size: 1232, extend_rcode: 0, version: 0, dnssec: false, edns_options: [Cookie(Cookie { client_cookie: [83, 150, 55, 76, 62, 1, 26, 211], server_cookie: None })] })] }
````

Note: 
1. when use 8.8.8.8, remove --get works fine
2. keep --get but use 1.1.1.1 works fine too.

Reference:

[google supports get method](https://developers.google.com/speed/public-dns/docs/doh)

**Comment by LinkTed** [comment]
I fix the bug in f5c52858428ce6268b60ffbe18f147e00b8ef7ff. The Google DNS server send a reset stream after the DNS response was transferred, which honest is kind of strange. Now, if the client has already received something and gets a reset stream, it considered that it got the whole DNS response, which should resolve the problem with the Google server. If you have some time, could you check if it works with the current master?

**Comment by cattyhouse** [comment]
@LinkTed the issue has been fixed, works great now, thank you.
